src/vmm/src/lib.rs

```diff
@@ -677,6 +686,8 @@ impl Vmm {
             huge_pages: None,
         };
 
+        self.resume_vm()?;
```

Comment:
While this get's the job done, it's not really what we should do here. `resume_vm()`
1. sends notifications to all devices on the MMIO bus
2. sends a "Resume" notification to _all_ vcpu threads
3. waits for the vcpu threads to get back to us with a "resumed" response

Here, we only need 2. and 3. for the newly added vcpus, but we probably don't need the device notifications, and we definitely don't want to send notifications to all other vcpus, because that just needlessly causes them to KVM_EXIT.

Reply:
Yes, I had noticed that, it was more just a convenient way to see if starting the threads would work as solution. I can probably create a new fn to do steps 2. and 3. and use that instead.

Reply:
Okay cool, I've now refactored it slightly, adding a new function `resume_vcpu_threads()`. It takes an int index as an argument and starts the threads starting after that. This should solve this problem